ci-operator/step-registry/ipi/deprovision/deprovision/in-bastion/ipi-deprovision-deprovision-in-bastion-commands.sh

```diff
@@ -0,0 +1,87 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+# Ensure our UID, which is randomly generated, is in /etc/passwd. This is required
+# to be able to SSH.
+if ! whoami &> /dev/null; then
+    if [[ -w /etc/passwd ]]; then
+        echo "${USER_NAME:-default}:x:$(id -u):0:${USER_NAME:-default} user:${HOME}:/sbin/nologin" >> /etc/passwd
+    else
+        echo "/etc/passwd is not writeable, and user matching this uid is not found."
+        exit 1
+    fi
+fi
+
+function run_command() {
+    local cmd="$1"
+    echo "Running Command: ${cmd}"
+    eval "${cmd}"
+}
+
+function run_ssh_cmd() {
+    local sshkey=$1
+    local user=$2
+    local host=$3
+    local remote_cmd=$4
+
+    options=" -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null "
+    cmd="ssh ${options} -i \"${sshkey}\" ${user}@${host} \"${remote_cmd}\""
+    run_command "$cmd" || return 2
+    return 0
+}
+
+function run_scp_from_remote() {
+    local sshkey=$1
+    local user=$2
+    local host=$3
+    local src=$4
+    local dest=$5
+
+    options=" -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null "
+    cmd="scp -r ${options} -i \"${sshkey}\" ${user}@${host}:${src} ${dest}"
+    run_command "$cmd" || return 2
+    return 0
+}
+
+function save_logs() {
+    echo "Copying the Installer logs and metadata to the artifacts directory..."
+    cp /tmp/installer/.openshift_install.log "${ARTIFACT_DIR}"
+    cp /tmp/installer/metadata.json "${ARTIFACT_DIR}"
+}
+
+trap 'CHILDREN=$(jobs -p); if test -n "${CHILDREN}"; then kill ${CHILDREN} && wait; fi' TERM
+trap 'save_logs' EXIT TERM
+
+REMOTE_DIR="/tmp"
+REMOTE_INSTALL_DIR="/tmp/installer/"
+REMOTE_ENV_FILE="/tmp/remote_env_file"
```

Comment:
Sounds we did not write anything into this file.

Reply:
file `remote_env_file` is already updated and uploaded into bastion host in step `ipi-install-install-in-bastion`, so here just only source it when running command in bastion host.

Reply:
got it.